src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -2267,7 +2267,7 @@ pub fn strip_debug(builder: &Builder<'_>, target: TargetSelection, path: &Path)
     // FIXME: to make things simpler for now, limit this to the host and target where we know
     // `strip -g` is both available and will fix the issue, i.e. on a x64 linux host that is not
     // cross-compiling. Expand this to other appropriate targets in the future.
-    if target != "x86_64-unknown-linux-gnu" || target != builder.config.build || !path.exists() {
+    if target != "x86_64-unknown-linux-gnu" || !builder.is_host_target(&target) || !path.exists() {
```

Comment:
For instance, I *think* this is actually trying to say `!builder.is_build_target(&target)`?

Reply:
What you think about `is_builder_target`?

Reply:
Seems reasonable, but maybe just is_build_target? Since we call it build platform right?

Reply:
I wanted to be very explicit. I feel like `is_build_target` might be confusing for some developers as it could mean either the platform being built for or the one being used to build.

Reply:
Right. I think its fine either way, as long as we are clear enough about this refers to the build platform not the host platform when build != host.